packages/dd-trace/src/appsec/waf/waf_context_wrapper.js

```diff
@@ -27,31 +28,39 @@ class WAFContextWrapper {
 
     const payload = {}
     let payloadHasData = false
-    const inputs = {}
     const newAddressesToSkip = new Set(this.addressesToSkip)
 
     if (persistent !== null && typeof persistent === 'object') {
-      // TODO: possible optimization: only send params that haven't already been sent with same value to this wafContext
+      const persistentInputs = {}
+
       for (const key of Object.keys(persistent)) {
-        // TODO: requiredAddresses is no longer used due to processor addresses are not included in the list. Check on
-        // future versions when the actual addresses are included in the 'loaded' section inside diagnostics.
-        if (!this.addressesToSkip.has(key)) {
-          inputs[key] = persistent[key]
+        if (!this.addressesToSkip.has(key) && this.knownAddresses.has(key)) {
+          persistentInputs[key] = persistent[key]
           if (preventDuplicateAddresses.has(key)) {
             newAddressesToSkip.add(key)
           }
```

Comment:
I don't get this check. `preventDuplicateAddresses` is a fixed Set, isn't it?

Reply:
So its values could be added when `newAddressesToSkip` is created in line 31.

Reply:
I don't understand your point...
> `preventDuplicateAddresses` is a fixed Set, isn't it?

Yes, it is. 

> So its values could be added when newAddressesToSkip is created in line 31

No, if we do that, instead of prevent sending it twice in the same request, we will send it only if we are trying to send it in the first call. We should add it to skipped Set only when a payload with the address happens.

Reply:
true, forget it :P